internal/controller/util_test.go

```diff
@@ -0,0 +1,98 @@
+package controller
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_createIngressRuleMatchFromURL(t *testing.T) {
+	type args struct {
+		url                     url.URL
+		includeLocalhost        bool
+		matchUnderscoreVersions bool
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{{
+		name: "no path",
+		args: args{
+			url:                     mustURLParse(t, "http://host.my"),
+			includeLocalhost:        false,
+			matchUnderscoreVersions: false,
+		},
+		want: "Host(`host.my`) && PathPrefix(``)",
+	}, {
+		name: "include localhost",
+		args: args{
+			url:                     mustURLParse(t, "http://host.my"),
+			includeLocalhost:        true,
+			matchUnderscoreVersions: false,
+		},
+		want: "(Host(`localhost`) || Host(`host.my`)) && PathPrefix(``)",
+	}, {
+		name: "no versions",
+		args: args{
+			url:                     mustURLParse(t, "http://host.my/some/path"),
+			includeLocalhost:        false,
+			matchUnderscoreVersions: false,
+		},
+		want: "Host(`host.my`) && PathPrefix(`/some/path`)",
```

Comment:
Minor ding maar geef de voorkeur aan gebruik van `example.com` aangezien dat een reserved FQDN is voor dit soort doeleinden (test/docs).